src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -150,5 +155,96 @@ public void ConnectHeaders_NewAndExistingHeaders_ShouldCreateNewHeaders()
             Assert.Equal(listOfNewHeaders.Last(), connectedHeadersNew.DownloadTo.Header);
             Assert.Equal(listOfNewHeaders.First(), connectedHeadersNew.DownloadFrom.Header);
         }
+
+        // Supply headers that we already have, make sure no new ChainedHeaders were created.
+        [Fact]
+        public void ConnectHeaders_SupplyExistingHeaders_DontCreateNewChainHeaders()
+        {
+            var testContext = new TestContext();
+            ChainedHeaderTree cht = testContext.CreateChainedHeaderTree();
+            ChainedHeader chainTip = testContext.ExtendAChain(5);
+            Assert.Empty(cht.GetChainedHeadersByHash());
+
+            cht.Initialize(chainTip, true);
+
+            Assert.Equal(6, cht.GetChainedHeadersByHash().Keys.Count);
+
+            var beforeKeys = new List<uint256>(cht.GetChainedHeadersByHash().Keys);
+
+            List<BlockHeader> listOfExistingHeaders = testContext.ChainedHeaderToList(chainTip, 5);
+            cht.ConnectNewHeaders(1, listOfExistingHeaders);
+            var afterKeys = new List<uint256>(cht.GetChainedHeadersByHash().Keys);
+
+            // check chainedHeadersByHash map before and after adding duplicate headers
+            Assert.Equal(beforeKeys.Count, afterKeys.Count);
+            Assert.True(beforeKeys.All(afterKeys.Contains));
+        }
+
+        // Supply some headers and after that supply some more headers
+        // and make sure that PeerTipsByHash is updated (the total amount of items is the same).
+        [Fact]
+        public void ConnectHeaders_SupplyHeadersThenSupplyMore_PeerTipsByHashShouldBeUpdated()
+        {
+            TestContext testContext = new TestContext();
+            ChainedHeaderTree cht = testContext.CreateChainedHeaderTree();
+
+            var chainTip = testContext.ExtendAChain(10);
```

Comment:
It would be much easier to read if the code style was respected.

Reply:
(not)using var :-)

Reply:
Corrected these

Reply:
Not really, you only done half where the type was not needed and you remove the type. You also need to do the other half where the type is required for example on line 168 and alike.

Reply:
Have amended as follows:  
Used var where return object type specified explicitly or implied by method call or return variable name (I've made names more descriptive where necessary).  For other cases where return is type is unknown or ambiguous have used explicit typing

Reply:
Reverted as per discussion.  Explicit typing used and var only where Type Name on RHS of =